x-pack/plugin/sql/build.gradle

```diff
@@ -39,6 +39,7 @@ check.dependsOn internalClusterTest
 
 dependencies {
   compileOnly project(path: xpackModule('core'), configuration: 'default')
+  compileOnly project(path: xpackModule('mapper-flattened'))
```

Comment:
Unfortunately this creates a runtime issue as plugins that are not marked as extensible (such as Painless) cannot have their classes imported into a different project.
In other words the `FlatObjectFieldMapper.CONTENT_TYPE` won't be available which would result in a CNFE.
As such as I would simply declare the String constant internally and use it verbatim instead of linking to it.